### PR TITLE
[MM-11287] Add support for plus sign and period/dot in custom URL schemes (#9155)

### PR DESCRIPTION
**Pull request by letsila**
Updates package.json and some unit tests after markdown parser [update](https://github.com/mattermost/marked/pull/11) (#9155)

[Jira ticket](https://mattermost.atlassian.net/browse/MM-11287)

#### Related Pull Requests

- Has server changes https://github.com/mattermost/mattermost-server/pull/10852
- Has marked library changes mattermost/marked#11
- Has commonmark.js library changes mattermost/commonmark.js#7

### CODE REVIEW

File: utils/text_formatting_links.test.jsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ describe('Markdown.Links', () => {
 
         assert.equal(
             Markdown.format('test.:test').trim(),
-            '<p>test.:test</p>'
+            '<p><a class="theme markdown__link" href="test.:test" rel="noreferrer" target="_blank">test.:test</a></p>'
```
Comment (hmhealey) [inline utils/text_formatting_links.test.jsx]: Could you move this to the `external links` section since we now consider this a link?

Comment (letsila) [reply utils/text_formatting_links.test.jsx]: Done

**Comment by letsila** [comment]
Fixed

**Comment by letsila** [comment]
@lieu-data sorry for that. I will avoid force push in the future.

**Pull request merged**